lib/scanner/csv-scanner/csv-scanner.c

```diff
@@ -376,7 +376,9 @@ csv_scanner_scan_next(CSVScanner *self)
 const gchar *
 csv_scanner_get_current_name(CSVScanner *self)
```

Comment:
With the original implementation there are only two cases where `csv_scanner_get_current_name` could not work:
 - There are more values than columns in the csv file, and we are not `CSV_SCANNER_GREEDY` . (In this case the `current_column` was setted to `NULL` during switch to next column, and we neglected the rest of the data.)
 - The other case was when there were absolutely no data at all in the line. But this was handled here https://github.com/balabit/syslog-ng/blob/da326364d52a21ebaf2d8fd17f4077fc5cc954e6/modules/csvparser/csvparser.c#L140
by the fact, that `csv_scanner_scan_next` returns false, and we stopped the parsing, and did not try to print any message with the `current_column`.

After the patch, `csv_scanner_get_current_name` will return wrong value in the first case. Fortunately we still do not print any message in this case. So practically it will not cause any problem. (Just CleanCode point of view.)

The second scenario - when the input is empty - is still valid, and the "selector" name is a pretty good error message for this. However I would prefer to handle the "no data" case by skipping the line. (This will also bring an another benefit: users may seperate their entry blocks with empty lines.)

Thank You @furiel for the investigation.

Reply:
I am not sure I understand everything here. I was handling the empty line case, but not in csv-parser() rather in add-contextual-data(), which has not handled empty lines, and I have not added that.

I agree It'd be a good idea to handle them there, as the csv file is indeed a configuration file, however as csv-parser() is intended to be used on incoming logs, we shouldn't specially handle empty lines.

And indeed, csv-parser() accepts empty lines, but I think it is a bug. 

I think we should discuss this IRL.

Reply:
Ssoo, I have added a refactor of csv-scanner so that it behaves more consistently and also addresses the things we discussed in this comment:

 * csv-parser() will drop empty lines instead of accepting them (this was a bug)
 * add-contextual-data() will ignore empty lines, instead of frowning on them (this is a feature)

At the same time, I have added a unit test at the csv-scanner level (earlier it was only tested through csv-parser()). This unit test is not yet complete, but at least shows the consistent usage of the API. Stuff like separators, and escaping modes are not tested here, rather they still are tested through csv-parser().

with that I consider this comment addressed. I am rewriting the summary of the PR so it better reflects what we have here.